core/src/main/scala/besom/internal/Config.scala

```diff
@@ -6,117 +6,64 @@ import scala.util.Try
 
 class Config private (
   val namespace: NonEmptyString,
-  private val isProjectName: Boolean,
-  private val configMap: Map[String, String],
-  private val configSecretKeys: Set[String]
+  private[internal] val isProjectName: Boolean,
+  private[internal] val configMap: Map[NonEmptyString, String],
+  private[internal] val configSecretKeys: Set[NonEmptyString]
 ):
-  private def fullKey(key: String): String = namespace + ":" + key
+  private def fullKey(key: NonEmptyString): NonEmptyString = namespace +++ ":" +++ key
```

Comment:
wouldn't it be more readable to just use string interpolation instead of `namespace +++ ":" +++ key`?

Reply:
That would probably require implementing an interpolator for nonempty strings